internal/telemetry/collector/local.go

```diff
@@ -0,0 +1,560 @@
+	prefix := "node.cpu.total:"
+	switch {
+	case cores <= 4:
+		return prefix + "0-4"
+	case cores <= 8:
+		return prefix + "4-8"
+	case cores <= 16:
+		return prefix + "8-16"
+	case cores <= 32:
+		return prefix + "16-32"
+	case cores <= 64:
+		return prefix + "32-64"
+	case cores <= 128:
+		return prefix + "64-128"
+	case cores <= 256:
+		return prefix + "128-256"
+	case cores <= 512:
+		return prefix + "256-512"
+	default:
+		return prefix + "512+"
+	}
+}
+
+func bucketMemoryGi(bytes uint64) string {
+	prefix := "node.memory.gibibytes:"
+	switch {
+	case bytes <= 16<<30:
+		return prefix + "0-16"
+	case bytes <= 32<<30:
+		return prefix + "16-32"
+	case bytes <= 64<<30:
+		return prefix + "32-64"
+	case bytes <= 128<<30:
+		return prefix + "64-128"
+	case bytes <= 256<<30:
+		return prefix + "128-256"
+	case bytes <= 512<<30:
+		return prefix + "256-512"
+	case bytes <= 1024<<30:
+		return prefix + "512-1024"
+	default:
+		return prefix + "1024+"
+	}
+}
+
+func bucketGPUTotal(gpu uint64) string {
+	prefix := "node.gpu.total:"
+	switch {
+	case gpu == 0:
+		return prefix + "0"
+	case gpu <= 2:
+		return prefix + "1-2"
+	case gpu <= 4:
+		return prefix + "3-4"
+	case gpu <= 8:
+		return prefix + "5-8"
+	case gpu <= 16:
+		return prefix + "9-16"
+	case gpu <= 32:
+		return prefix + "17-32"
+	case gpu <= 64:
+		return prefix + "33-64"
+	case gpu <= 128:
+		return prefix + "65-128"
+	case gpu <= 256:
+		return prefix + "129-256"
+	default:
+		return prefix + "257+"
+	}
+}
+
+var kubeVersionBuckets = map[string]struct{}{
+	"1.29": {},
+	"1.30": {},
+	"1.31": {},
+	"1.32": {},
+	"1.33": {},
+	"1.34": {},
+	"1.35": {},
+	"1.36": {},
+	"1.37": {},
+}
+
+func bucketKubeVersion(v string) string {
+	if _, ok := kubeVersionBuckets[v]; ok {
+		return "node.info.kubeVersion:" + v
+	}
+	return "node.info.kubeVersion:other"
+}
+
+func bucketOS(osname string) string {
+	prefix := "node.info.os:"
+	switch v := strings.ToLower(osname); v {
+	case "linux", "windows":
+		return prefix + v
+	default:
+		return prefix + "other"
+	}
+}
+
+func bucketArch(arch string) string {
+	prefix := "node.info.arch:"
+	switch v := strings.ToLower(arch); v {
+	case "amd64", "arm64":
+		return prefix + v
+	default:
+		return prefix + "other"
+	}
+}
+
+func bucketProvider(p string) string {
+	// (bootstrap|control-plane|infrastructure)-<provider>
+	// NOTE: very opinionated metric, let us reflect whether we actually need it
+	return "providers:" + p
+}
+
+func bucketOperatorInstalled(typ string, v bool) string {
+	return "gpu.operator_installed." + typ + ":" + strconv.FormatBool(v)
+}
+
+func bucketSyncMode(mode string) string {
+	return "syncMode:" + strings.ToLower(mode)
+}
+
+func bucketPodsWithGPURequests(n uint64) string {
+	prefix := "pods.with_gpu_requests:"
+	switch {
+	case n == 0:
+		return prefix + "0"
+	case n <= 4:
+		return prefix + "1-4"
+	case n <= 16:
+		return prefix + "5-16"
+	case n <= 64:
+		return prefix + "17-64"
+	case n <= 256:
+		return prefix + "64-256"
```

Comment:
```suggestion
		return prefix + "65-256"
```

Reply:
[Done](https://github.com/k0rdent/kcm/compare/422b405bee6ab4a935e32bc4c57cb23ff7ab2f77..9869ec47c692fa71cb8265533f5d2b72e9b8055e) (and expanded), thanks

Reply:
The approach has been reworked to automate ranges creation